datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -532,66 +521,131 @@ impl PhysicalExpr for InListExpr {
             match value_data_type {
                 DataType::Boolean => {
                     let array = array.as_any().downcast_ref::<BooleanArray>().unwrap();
-                    set_contains_with_negated!(array, set, self.negated)
+                    Ok(set_contains_for_primitive!(
```

Comment:
I'm wondering why `set_contains_for_primitive` is macro, but `set_contains_utf` and `make_set_contains_decimal` are methods?

Reply:
`set_contains_utf` and `make_set_contains_decimal` are just apply to a specified data type case which are not compatible with other primitive case.
`set_contains_for_primitive` will apply to many similar case and same logic